kore/src/Kore/Step/Simplification/SubstitutionSimplifier.hs

```diff
@@ -107,14 +108,15 @@ import Kore.Variables.UnifiedVariable
     , isSetVar
     )
 
-newtype SubstitutionSimplifier simplifier =
+newtype SubstitutionSimplifier simplifier1 simplifier2 =
     SubstitutionSimplifier
         { simplifySubstitution
             :: forall variable
             .  InternalVariable variable
-            => SideCondition variable
+            => NotSimplifier simplifier1
```

Comment:
This argument should not be here. If `SubstitutionSimplifier` needs a `NotSimplifier`, then it should capture it when it is created, i.e. the `NotSimplifier` should be an argument to `substitutionSimplifier`.